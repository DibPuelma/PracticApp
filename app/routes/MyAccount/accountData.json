{
  "name": "Agust√≠n",
  "lastName": "Ibieta",
  "email": "aibieta3@usm.cl",
  "points": 150,
  "level": 2,
  "totalAverage": 3.6,
  "pollsAnswered": 10,
  "prizesWon": 2,
  "storesHelped": 7,
  "brandsHelped": 5
}
