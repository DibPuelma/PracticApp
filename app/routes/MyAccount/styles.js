import {
  StyleSheet,
  Dimensions
} from 'react-native';

module.exports = StyleSheet.create({
  background: {
    flex: 1,
    marginLeft: 10,
    marginTop: 72,
    marginRight: 10,
    marginBottom: 10,
    elevation: 3,
    backgroundColor: '#E6E6E6',
    borderColor: '#C6C6C6',
    borderWidth: 1
  },
  container: {
    justifyContent: 'center',
    flexWrap: 'wrap',
    backgroundColor: '#E6E6E6',
  },
  userData: {
    marginTop: 20,
    alignItems: 'center'
  },
  name: {
    fontSize: 20
  },
  email: {
    fontSize: 18,
    fontStyle: 'italic'
  },
  badgesContainer: {
    marginTop: 20,
    height: 300,
    flexWrap: 'wrap'
  },
  rowOfBadges:Â {
    flexDirection: 'row',
    flex: 3,
  },
  badge: {
    flex: 1,
    margin: 8,
    alignItems: 'center',
    justifyContent: 'center',
    elevation: 4,
    borderWidth: 1,
    borderColor: '#C6C6C6',
    backgroundColor: '#f84a4a',
    borderRadius: 8
  },
  normalText: {
    textAlign: 'center',
    fontSize: 16,
    marginBottom: 5
  }
});
