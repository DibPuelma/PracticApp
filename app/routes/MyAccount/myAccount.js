import React, { Component } from 'react';
import {
  Image,
  Text,
  View,
  ScrollView
} from 'react-native';

import accountData from './accountData';
import backButtonHandler from '../../lib/backButtonHandler';
import styles from './styles';

export default class EvaluationDetails extends Component {
  constructor(props){
    super(props);
    this.singletonBackButtonHandler = backButtonHandler.getInstance();
    this._backToPrevious = this._backToPrevious.bind(this);
  }

  componentWillMount(){
    this.singletonBackButtonHandler.addBackEvent(this._backToPrevious);
  }

  componentWillUnmount() {
    this.singletonBackButtonHandler.removeBackEvent(this._backToPrevious);
  }

  _backToPrevious() {
    this.props.navigator.replace({id: 'QRReader'});
    return true; // This is important to prevent multiple calls
  }

  render(){
    return(
      <View style={styles.background}>
      <ScrollView contentContainerStyle={styles.container}>
      <View style={styles.userData}>
      <Text style={styles.name}> Hola {accountData.name} {accountData.lastName}</Text>
      <Text style={styles.email}> {accountData.email} </Text>
      </View>
      <View style={styles.badgesContainer}>

      <View style={styles.rowOfBadges}>
      <View style={styles.badge}>
      <Text style={styles.normalText}> Puntos </Text>
      <Text style={styles.normalText}> {accountData.points} </Text>
      </View>

      <View style={styles.badge}>
      <Text style={styles.normalText}> Evaluaci√≥n Promedio </Text>
      <Text style={styles.normalText}> {accountData.totalAverage} </Text>
      </View>

      <View style={styles.badge}>
      <Text style={styles.normalText}> Premios Ganados </Text>
      <Text style={styles.normalText}> {accountData.prizesWon} </Text>
      </View>
      </View>

      <View style={styles.rowOfBadges}>
      <View style={styles.badge}>
      <Text style={styles.normalText}> Encuestas contestadas </Text>
      <Text style={styles.normalText}> {accountData.pollsAnswered} </Text>
      </View>

      <View style={styles.badge}>
      <Text style={styles.normalText}> Tiendas ayudadas </Text>
      <Text style={styles.normalText}> {accountData.storesHelped} </Text>
      </View>

      <View style={styles.badge}>
      <Text style={styles.normalText}> Marcas Ayudadas </Text>
      <Text style={styles.normalText}> {accountData.brandsHelped} </Text>
      </View>
      </View>

      <View style={styles.badge}>
      <Text style={styles.normalText}> Nivel </Text>
      <Text style={styles.normalText}> {accountData.level} </Text>
      </View>

      </View>
      </ScrollView>
      </View>
    );
  }
}
