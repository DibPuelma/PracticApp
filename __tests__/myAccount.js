import 'react-native';
import React from 'react';
import MyAccount from '../app/routes/MyAccount/myAccount'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <MyAccount />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
