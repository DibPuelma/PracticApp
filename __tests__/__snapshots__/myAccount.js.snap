exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#E6E6E6",
      "borderColor": "#C6C6C6",
      "borderWidth": 1,
      "elevation": 3,
      "flex": 1,
      "marginBottom": 10,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 72,
    }
  }>
  <ScrollView
    contentContainerStyle={
      Object {
        "backgroundColor": "#E6E6E6",
        "flexWrap": "wrap",
        "justifyContent": "center",
      }
    }>
    <View
      style={
        Object {
          "alignItems": "center",
          "marginTop": 20,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
          }
        }>
         Hola 
        Agustín
         
        Ibieta
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 18,
            "fontStyle": "italic",
          }
        }>
         
        aibieta3@usm.cl
         
      </Text>
    </View>
    <View
      style={
        Object {
          "flexWrap": "wrap",
          "height": 300,
          "marginTop": 20,
        }
      }>
      <View
        style={
          Object {
            "flex": 3,
            "flexDirection": "row",
          }
        }>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#f84a4a",
              "borderColor": "#C6C6C6",
              "borderRadius": 8,
              "borderWidth": 1,
              "elevation": 4,
              "flex": 1,
              "justifyContent": "center",
              "margin": 8,
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 16,
                "marginBottom": 5,
                "textAlign": "center",
              }
            }>
             Puntos 
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 16,
                "marginBottom": 5,
                "textAlign": "center",
              }
            }>
             
            150
             
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#f84a4a",
              "borderColor": "#C6C6C6",
              "borderRadius": 8,
              "borderWidth": 1,
              "elevation": 4,
              "flex": 1,
              "justifyContent": "center",
              "margin": 8,
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 16,
                "marginBottom": 5,
                "textAlign": "center",
              }
            }>
             Evaluación Promedio 
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 16,
                "marginBottom": 5,
                "textAlign": "center",
              }
            }>
             
            3.6
             
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#f84a4a",
              "borderColor": "#C6C6C6",
              "borderRadius": 8,
              "borderWidth": 1,
              "elevation": 4,
              "flex": 1,
              "justifyContent": "center",
              "margin": 8,
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 16,
                "marginBottom": 5,
                "textAlign": "center",
              }
            }>
             Premios Ganados 
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 16,
                "marginBottom": 5,
                "textAlign": "center",
              }
            }>
             
            2
             
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 3,
            "flexDirection": "row",
          }
        }>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#f84a4a",
              "borderColor": "#C6C6C6",
              "borderRadius": 8,
              "borderWidth": 1,
              "elevation": 4,
              "flex": 1,
              "justifyContent": "center",
              "margin": 8,
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 16,
                "marginBottom": 5,
                "textAlign": "center",
              }
            }>
             Encuestas contestadas 
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 16,
                "marginBottom": 5,
                "textAlign": "center",
              }
            }>
             
            10
             
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#f84a4a",
              "borderColor": "#C6C6C6",
              "borderRadius": 8,
              "borderWidth": 1,
              "elevation": 4,
              "flex": 1,
              "justifyContent": "center",
              "margin": 8,
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 16,
                "marginBottom": 5,
                "textAlign": "center",
              }
            }>
             Tiendas ayudadas 
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 16,
                "marginBottom": 5,
                "textAlign": "center",
              }
            }>
             
            7
             
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#f84a4a",
              "borderColor": "#C6C6C6",
              "borderRadius": 8,
              "borderWidth": 1,
              "elevation": 4,
              "flex": 1,
              "justifyContent": "center",
              "margin": 8,
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 16,
                "marginBottom": 5,
                "textAlign": "center",
              }
            }>
             Marcas Ayudadas 
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 16,
                "marginBottom": 5,
                "textAlign": "center",
              }
            }>
             
            5
             
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#f84a4a",
            "borderColor": "#C6C6C6",
            "borderRadius": 8,
            "borderWidth": 1,
            "elevation": 4,
            "flex": 1,
            "justifyContent": "center",
            "margin": 8,
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 16,
              "marginBottom": 5,
              "textAlign": "center",
            }
          }>
           Nivel 
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 16,
              "marginBottom": 5,
              "textAlign": "center",
            }
          }>
           
          2
           
        </Text>
      </View>
    </View>
  </ScrollView>
</View>
`;
